### PR TITLE
feat(HACBS-1375): drop Application Prefix from API names

### PR DESCRIPTION
**Pull request by dirgim**
* Modify all application-api Application* CRs

Signed-off-by: dirgim <kpavic@redhat.com>

### CODE REVIEW

**Review by sonam1412 — APPROVED** [review]
LGTM

**Review by jsztuka — APPROVED** [review]
lgtm

**Comment by jsztuka** [comment]
Do we have story for updating this relation-diagram ?
https://issues.redhat.com/secure/attachment/12728398/HACBS%20Test%20Workflow%20-%20Integration%20Service.png

**Comment by dirgim** [comment]
Just realized that we'll need to wait for https://github.com/redhat-appstudio/release-service/pull/122 to be merged before merging this one.

**Comment by hongweiliu17** [comment]
@dirgim LGTM. Could you also update the prefix in https://github.com/redhat-appstudio/integration-service/blob/main/config/samples/kcp/kcp_test_resource.yaml and https://github.com/redhat-appstudio/integration-service/blob/main/config/samples/kcp/README.md or delete folder https://github.com/redhat-appstudio/integration-service/tree/main/config/samples/kcp directly if we don't need the kcp example any more, thank you!

**Review by hongweiliu17 — APPROVED** [review]

File: controllers/snapshot/snapshot_adapter.go [hunk 1]
```diff
@@ -276,27 +276,27 @@ func (a *Adapter) createMissingReleasesForReleasePlans(application *applicationa
...
 		if existingRelease != nil {
 			a.logger.Info("Found existing Release",
-				"ApplicationSnapshot.Name", applicationSnapshot.Name,
+				"Snapshot.Name", snapshot.Name,
 				"ReleasePlan.Name", releasePlan.Name,
 				"Release.Name", existingRelease.Name)
 		} else {
-			newRelease := release.CreateReleaseForReleasePlan(&releasePlan, applicationSnapshot)
+			newRelease := release.CreateReleaseForReleasePlan(&releasePlan, snapshot)
 			// copy PipelineRun PAC annotations/labels from applicaitonSnapshot to Release
```
Comment (johnbieren) [inline controllers/snapshot/snapshot_adapter.go:284]: `s/applicaitonSnapshot/snapshot/`

File: controllers/snapshot/snapshot_adapter.go [hunk 2]
```diff
@@ -372,33 +372,33 @@ func (a *Adapter) getAllApplicationComponents(application *applicationapiv1alpha
...
-func (a *Adapter) createIntegrationPipelineRun(application *applicationapiv1alpha1.Application, integrationTestScenario *v1alpha1.IntegrationTestScenario, applicationSnapshot *applicationapiv1alpha1.ApplicationSnapshot) error {
-	pipelineRun := tekton.NewIntegrationPipelineRun(applicationSnapshot.Name, application.Namespace, *integrationTestScenario).
-		WithApplicationSnapshot(applicationSnapshot).
+func (a *Adapter) createIntegrationPipelineRun(application *applicationapiv1alpha1.Application, integrationTestScenario *v1alpha1.IntegrationTestScenario, snapshot *applicationapiv1alpha1.Snapshot) error {
+	pipelineRun := tekton.NewIntegrationPipelineRun(snapshot.Name, application.Namespace, *integrationTestScenario).
+		WithSnapshot(snapshot).
 		WithIntegrationLabels(integrationTestScenario).
 		WithApplicationAndComponent(a.application, a.component).
 		AsPipelineRun()
 	// copy PipelineRun PAC annotations/labels from applicaitonSnapshot to integration test PipelineRuns
```
Comment (johnbieren) [inline controllers/snapshot/snapshot_adapter.go:383]: `s/applicaitonSnapshot/snapshot/`

File: gitops/gitops_test.go [hunk 3]
```diff
@@ -143,10 +143,10 @@ var _ = Describe("Gitops functions for managing ApplicationSnapshots", Ordered,
 		Expect(gitops.HaveHACBSTestsSucceeded(hasSnapshot)).To(BeTrue())
```
Comment (johnbieren) [inline gitops/gitops_test.go:143]: It won't let me comment on it for some reason, but line 141 `s/applicationsnapshot/snapshot/`

**Review by johnbieren — CHANGES_REQUESTED** [review]
few small ones to fix

File: controllers/snapshot/cache.go [hunk 4]
```diff
@@ -34,32 +34,32 @@ func SetupReleasePlanCache(mgr ctrl.Manager) error {
...
-		return []string{obj.(*releasev1alpha1.Release).Spec.ApplicationSnapshot}
+		return []string{obj.(*releasev1alpha1.Release).Spec.Snapshot}
 	}
 
 	return mgr.GetCache().IndexField(context.Background(), &releasev1alpha1.Release{},
-		"spec.applicationSnapshot", releaseIndexFunc)
+		"spec.snapshot", releaseIndexFunc)
 }
 
 // SetupApplicationCache adds a new index field to be able to search Applications by Environment.
```
[the remark this answers is not in the document]
Comment (dirgim) [reply controllers/snapshot/cache.go]: Good point, fixed

File: controllers/snapshot/cache.go [hunk 5]
```diff
@@ -34,32 +34,32 @@ func SetupReleasePlanCache(mgr ctrl.Manager) error {
...
-		return []string{obj.(*applicationapiv1alpha1.ApplicationSnapshotEnvironmentBinding).Spec.Environment}
+		return []string{obj.(*applicationapiv1alpha1.SnapshotEnvironmentBinding).Spec.Environment}
 	}
 
-	return mgr.GetCache().IndexField(context.Background(), &applicationapiv1alpha1.ApplicationSnapshotEnvironmentBinding{},
+	return mgr.GetCache().IndexField(context.Background(), &applicationapiv1alpha1.SnapshotEnvironmentBinding{},
 		"spec.environment", applicationIndexFunc)
 }
 
 // SetupEnvironmentCache adds a new index field to be able to search Environments by Application.
```
[the remark this answers is not in the document]
Comment (dirgim) [reply controllers/snapshot/cache.go]: Ack, fixed

File: gitops/binding.go [hunk 6]
```diff
@@ -24,28 +24,28 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// CreateApplicationSnapshotEnvironmentBinding creates a new ApplicationSnapshotEnvironmentBinding using the provided info.
-func CreateApplicationSnapshotEnvironmentBinding(bindingName string, namespace string, applicationName string, environmentName string, appSnapshot *applicationapiv1alpha1.ApplicationSnapshot, components []applicationapiv1alpha1.Component) *applicationapiv1alpha1.ApplicationSnapshotEnvironmentBinding {
+// CreateSnapshotEnvironmentBinding creates a new SnapshotEnvironmentBinding using the provided info.
+func CreateSnapshotEnvironmentBinding(bindingName string, namespace string, applicationName string, environmentName string, appSnapshot *applicationapiv1alpha1.Snapshot, components []applicationapiv1alpha1.Component) *applicationapiv1alpha1.SnapshotEnvironmentBinding {
```
[the remark this answers is not in the document]
Comment (dirgim) [reply gitops/binding.go]: Good catch, fixed

File: gitops/binding.go [hunk 7]
```diff
@@ -59,21 +59,21 @@ func CreateBindingComponents(components []applicationapiv1alpha1.Component) *[]a
 	return &bindingComponents
 }
 
-// FindExistingApplicationSnapshotEnvironmentBinding attempts to find an ApplicationSnapshotEnvironmentBinding that's
+// FindExistingSnapshotEnvironmentBinding attempts to find an SnapshotEnvironmentBinding that's
```
[the remark this answers is not in the document]
Comment (dirgim) [reply gitops/binding.go]: Always have been terrible at an vs a so didn't even notice :) Fixed, thanks

File: gitops/snapshot.go [hunk 8]
```diff
@@ -199,7 +199,7 @@ func CompareApplicationSnapshots(expectedApplicationSnapshot *applicationapiv1al
 	return true
 }
 
-// IsSnapshotCreatedByPushEvent checks if an applicationSnapshot has label PipelineAsCodeEventTypeLabel and with push value
-func IsSnapshotCreatedByPushEvent(applicationSnapshot *applicationapiv1alpha1.ApplicationSnapshot) bool {
-	return helpers.HasLabelWithValue(applicationSnapshot, PipelineAsCodeEventTypeLabel, PipelineAsCodePushType)
+// IsSnapshotCreatedByPushEvent checks if an snapshot has label PipelineAsCodeEventTypeLabel and with push value
```
[the remark this answers is not in the document]
Comment (dirgim) [reply gitops/snapshot.go]: Fixed, thanks

File: controllers/snapshot/snapshot_adapter.go [hunk 9]
```diff
@@ -372,33 +372,33 @@ func (a *Adapter) getAllApplicationComponents(application *applicationapiv1alpha
 }
 
 // createIntegrationPipelineRun creates and returns a new integration PipelineRun. The Pipeline information and the parameters to it
-// will be extracted from the given integrationScenario. The integration's ApplicationSnapshot will also be passed to the
+// will be extracted from the given integrationScenario. The integration's Snapshot will also be passed to the
```
[the remark this answers is not in the document]
Comment (dirgim) [reply controllers/snapshot/snapshot_adapter.go]: Good catch, fixed

File: controllers/snapshot/snapshot_adapter.go (hunk 1 above)
Comment (dirgim) [reply controllers/snapshot/snapshot_adapter.go]: Ack, fixed

File: controllers/snapshot/snapshot_adapter.go (hunk 2 above)
Comment (dirgim) [reply controllers/snapshot/snapshot_adapter.go]: Ack, fixed

File: gitops/gitops_test.go (hunk 3 above)
Comment (dirgim) [reply gitops/gitops_test.go:143]: Ack, fixed

**Review by johnbieren — APPROVED** [review]

**Pull request merged**